File: Assignment_27/program1.c
// Write a program which accept string from user and copy the contents of that string into another string. (Implement strcpy() function)

///////////////////////////////////////////////////////////////////////////////
// File Name   :  program1.c
// Description :  Accept string from user and  copy the contents of that string into another string. 
// Autor       :  Ankita Anil Patil
// Date        :  06/06/2025
///////////////////////////////////////////////////////////////////////////////



/*
    Input : "Marvellous Multi OS"

    Output : "Marvellous Multi OS"  in another string
*/

#include<stdio.h>

void StrCpyX(char *src, char *dest)
{
    if(src ==NULL)
    {
        printf("Invalid input ");
        return -1;
    }
    while(*src != '\0')
    {
        *dest = *src;
        src++;
        dest++;
    }
    *dest ='\0';
}

int main()
{
    char arr[30] = "Marvellous Multi OS";
    char brr[30];

    StrCpyX(arr,brr);

    printf("Copied string is : %s\n",brr);

    return 0;
}

File: Assignment_51/program1.java
// Write a java program which accept matrix from user and swap the contents of each consecutive rows.

/*
 
    Input  Matrix is :
        1 4 7 5
        8 9 2 4
        1 5 6 3
        4 7 5 6

    Updated Matrix is :
        8 9 2 4
        1 4 7 5
        4 7 5 6
        1 5 6 3
  
 */

  
///////////////////////////////////////////////////////////////////////////////
// File Name   :  program1.java
// Description :  Accept matrix from user and swap the contents of each consecutive rows.          
// Autor       :  Ankita Anil Patil
// Date        :  19/08/2025
///////////////////////////////////////////////////////////////////////////////
 
import java.util.Scanner;

class Matrix 
{ 
    int Arr[][];
   
    public Matrix(int iRow, int iCol) 
    { 
       Arr = new int [iRow][iCol];     
    } 

    public void Accept()
    {
        int i = 0, j = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Values in matrix : ");

        
        for(i = 0; i < Arr.length; i++)
        {
           for(j = 0; j < Arr[0].length; j++)
           {
                 Arr[i][j] = sobj.nextInt();
           }
                
           
        }

    }

    public void  SwapMatrixByRow()
    {
        int i = 0, j = 0, temp = 0;

        
        for(i = 0; i < Arr.length -1; i += 2)
        {
           for(j = 0; j < Arr[0].length; j++)
           {
                temp = Arr[i][j];
                Arr[i][j] =  Arr[i + 1][j];
                Arr[i + 1][j] = temp;
           }
        
           System.out.println();   
        }
        
       

    }

    public void Display()
    {
        int i = 0, j = 0;

        System.out.println(" Matrix is : ");

        
        for(i = 0; i < Arr.length; i++)
        {
           for(j = 0; j < Arr[0].length; j++)
           {
                System.out.print (Arr[i][j] + " "); 
           }
        
           System.out.println();   
        }
    }
}

class program1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of rows : ");
        int iValue1 = sobj.nextInt();

        System.out.println("Enter number of columns : ");
        int iValue2 = sobj.nextInt();

        Matrix pobj = new Matrix(iValue1,iValue2);

        pobj.Accept();
        pobj.Display();
        pobj.SwapMatrixByRow();
        pobj.Display();
        

    }
}